tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/Lambda/HandlerWrapperSetHandlerIntegration.cs

```diff
@@ -0,0 +1,130 @@
+    {
+        public bool PreserveAsyncContext => false;
+
+        public object OnDelegateBegin<TArg1>(object sender, ref TArg1 arg)
+        {
+            LambdaCommon.Log("DelegateWrapper Running OnDelegateBegin");
+
+            Scope scope;
+            var proxyInstance = arg.DuckCast<IInvocationRequest>();
+            if (proxyInstance == null)
+            {
+                LambdaCommon.Log("DuckCast.IInvocationRequest got null proxyInstance", debug: false);
+                scope = LambdaCommon.SendStartInvocation(new LambdaRequestBuilder(), string.Empty, null);
+            }
+            else
+            {
+                var jsonString = ConvertPayloadStream(proxyInstance.InputStream);
+                scope = LambdaCommon.SendStartInvocation(new LambdaRequestBuilder(), jsonString, proxyInstance.LambdaContext?.ClientContext?.Custom);
+            }
+
+            LambdaCommon.Log("DelegateWrapper FINISHED Running OnDelegateBegin");
+            return new CallTargetState(scope);
+        }
+
+        public void OnException(object sender, Exception ex)
+        {
+            LambdaCommon.Log("OnDelegateBegin could not send payload to the extension", ex, false);
+        }
+
+        public TReturn OnDelegateEnd<TReturn>(object sender, TReturn returnValue, Exception exception, object state)
+        {
+            // Needed in order to make this Async1Callbacks work with Func1Wrapper, which expects IReturnCallback
+            return returnValue;
+        }
+
+        /// <inheritdoc/>
+        public Task<TInnerReturn> OnDelegateEndAsync<TInnerReturn>(object sender, TInnerReturn returnValue, Exception exception, object state)
+        {
+            LambdaCommon.Log("DelegateWrapper Running OnDelegateEndAsync");
+            try
+            {
+                var proxyInstance = returnValue.DuckCast<IInvocationResponse>();
+                if (proxyInstance == null)
+                {
+                    LambdaCommon.Log("DuckCast.IInvocationResponse got null proxyInstance", debug: false);
+                    LambdaCommon.EndInvocationAsync(string.Empty, exception, ((CallTargetState)state!).Scope, RequestBuilder).ConfigureAwait(false);
+                }
+                else
+                {
+                    var jsonString = ConvertPayloadStream(proxyInstance.OutputStream);
+                    LambdaCommon.EndInvocationAsync(jsonString, exception, ((CallTargetState)state!).Scope, RequestBuilder).ConfigureAwait(false);
+                }
+            }
+            catch (Exception ex)
+            {
+                LambdaCommon.Log("OnDelegateEndAsync could not send payload to the extension", ex, false);
+                LambdaCommon.EndInvocationAsync(string.Empty, ex, ((CallTargetState)state!).Scope, RequestBuilder).ConfigureAwait(false);
+            }
+
+            LambdaCommon.Log("DelegateWrapper FINISHED Running OnDelegateEndAsync");
+            return Task.FromResult(returnValue);
```

Comment:
If the calls to `EndInvocationAsync` are awaited then you don't need `Task.FromResult`

Reply:
https://github.com/DataDog/dd-trace-dotnet/pull/4752/